src/odoc/bin/main.ml

```diff
@@ -262,6 +262,17 @@ end = struct
         & opt (some string) None
         & info ~docs ~docv:"PARENT" ~doc [ "parent" ])
     in
+    let search_asset =
+      let doc =
+        "A script that needs to be loaded for search to work. Will be looked \
+         up as an asset of the current compilation unit or one of its parent. \
+         When this argument is present multiple times, the scripts will be \
+         loaded in the same order as in the command line."
+      in
+      Arg.(
+        value & opt_all string []
+        & info ~docs ~docv:"ASSET" ~doc [ "search-asset" ])
```

Comment:
Since the search asset argument is a reference, and references are resolved at the link stage, this needs to be passed earlier than the html-generate phase, no? We would not have the ability to resolve them as we need to lookup the parents.
Did I miss something?

Reply:
That's right, units should have a list of the resolved assets of their parents with my suggestion. Isn't that the case already?

Reply:
Indeed we cannot resolve every possible assets and search assets have a special meaning and must be specified. And since 73e10e9 there's nothing to improve.